app/gui/src/main/scala/org/bitcoins/gui/ContractGUI.scala

```diff
@@ -0,0 +1,274 @@
+  MenuItem,
+  TableColumn,
+  TableView,
+  TextField
+}
+import scalafx.scene.layout._
+
+import java.io.File
+import java.nio.file.Files
+import java.text.SimpleDateFormat
+import java.util.Date
+import scala.util.{Failure, Success}
+
+class ContractGUI(glassPane: VBox) {
+
+  def fetchStartingData(): Unit = {
+    // Nothing to fetch for state yet...
+  }
+
+  private[gui] lazy val model = new ContractGUIModel()
+
+  private lazy val addEventTF = new TextField {
+    styleClass += "title-textfield"
+    promptText = "New Event Hex"
+    onKeyTyped = _ => {
+      val event = model.addEvent(this.text.value.trim)
+      event match {
+        case Some(tup) =>
+          clearEventTF()
+          eventTableView.sort()
+          // Set focus on new item
+          eventTableView.getSelectionModel().select(tup)
+          // Show view
+          showCreateOfferPane(tup._1, tup._2)
+        case None =>
+      }
+    }
+  }
+
+  private def clearEventTF(): Unit = {
+    addEventTF.clear()
+  }
+
+  lazy val addEventHBox = new HBox {
+    styleClass += "small"
+    children = Seq(addEventTF)
+  }
+
+  private lazy val addContractTF = new TextField {
+    styleClass += "title-textfield"
+    promptText = "Contract Hex"
+    onKeyTyped = _ => {
+      val bool = onContractAdded(text.value.trim, None)
+      if (bool) clearContractTF() // Clear on valid data
+      ()
+    }
+  }
+
+  private def clearContractTF(): Unit = {
+    addContractTF.clear()
+  }
+
+  private lazy val fileChooserButton = GUIUtil.getFileChooserButton(file => {
+    val hex = Files.readAllLines(file.toPath).get(0)
+    val bool = onContractAdded(hex, Some(file))
+    if (bool) addContractTF.clear()
+  })
+
+  lazy val addContractHBox = new HBox {
+    styleClass += "small"
+    children = Seq(fileChooserButton, addContractTF)
+  }
+
+  private lazy val eventIdCol = new TableColumn[
+    (OracleAnnouncementV0TLV, Option[ContractInfoV0TLV]),
+    String] {
+    text = "Event Id"
+    prefWidth = 160
+    cellValueFactory = { status =>
+      val eventIdStr = status.value._1.eventTLV.eventId
+      new StringProperty(status, "Event Id", eventIdStr)
+    }
+  }
+
+  private val ISO_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'")
+
+  private lazy val maturityCol = new TableColumn[
+    (OracleAnnouncementV0TLV, Option[ContractInfoV0TLV]),
+    String] {
+    text = "Matures"
+    prefWidth = 130
+    cellValueFactory = { status =>
+      val d = new Date(
+        status.value._1.eventTLV.eventMaturityEpoch.toLong * 1000)
+      new StringProperty(status, "Matures", ISO_FORMAT.format(d))
+    }
+  }
+
+  private lazy val contractCol = new TableColumn[
+    (OracleAnnouncementV0TLV, Option[ContractInfoV0TLV]),
+    String] {
+    text = "Contract"
+    prefWidth = 160
+    cellValueFactory = { status =>
+      val contractStr =
+        status.value._2 match {
+          case None        => ""
+          case Some(value) => value.contractDescriptor.toString()
```

Comment:
should use `_.hex` here instead of `toString`

Reply:
I used toString because I wanted to see the JSON structure. I don't know that either are really worth presenting to the user in string form.